### ISSUE TITLE
Search returning null features for searches with no results

### ISSUE THREAD

**Issue by lossyrob** [stac-utils/pgstac#45]
Upgrading stac-fastapi to pgstac 0.3.2, some tests are failing because search returns look like:

```javascript
{
  "type": "FeatureCollection",
  "context": {
    "limit": 10,
    "matched": 0,
    "returned": 0
  },
  "features": null
}
```

Seems as though pgstac is returning `null` instead of `[]` for empty searches, at least in some cases.

**Comment by vincentsarago** [comment]
I think this is due to `fields` 

```sql
select s from search('{"collections":["something"]}') s;
>>> {"next": null, "prev": null, "type": "FeatureCollection", "context": {"limit": 10, "matched": 0, "returned": 0}, "features": []}

select s from search('{"collections":["something"],"fields":{"include":["id"]}}') s;
>>>  {"next": null, "prev": null, "type": "FeatureCollection", "context": {"limit": 10, "matched": 0, "returned": 0}, "features": null}
```

@lossyrob could you point out which tests were failing ?

**Comment by lossyrob** [comment]
These tests:

```
FAILED clients/test_postgres.py::test_delete_item - TypeError: 'NoneType' object is not iterable
FAILED resources/test_item.py::test_delete_item - TypeError: 'NoneType' object is not iterable
FAILED resources/test_item.py::test_item_search_by_id_no_results_post - TypeError: 'NoneType' ...
FAILED resources/test_item.py::test_item_search_get_query_extension - TypeError: 'NoneType' ob...
```

**Comment by vincentsarago** [comment]
fixed in #46
